cmd/calyptia/get_trace_session.go

```diff
@@ -0,0 +1,250 @@
+	fs.BoolVar(&showID, "show-id", false, "Show trace session ID. Only applies when output format is table")
+	fs.StringVarP(&outputFormat, "output-format", "o", "table", "Output format. Allowed: table, json, yaml")
+
+	_ = cmd.RegisterFlagCompletionFunc("output-format", config.completeOutputFormat)
+
+	_ = cmd.RegisterFlagCompletionFunc("pipeline", config.completePipelines)
+
+	return cmd
+}
+
+func renderTraceSessionsTable(w io.Writer, ss types.TraceSessions, pipelineID string, showIDs bool) error {
+	tw := tabwriter.NewWriter(w, 0, 4, 1, ' ', 0)
+	if showIDs {
+		if _, err := fmt.Fprint(tw, "ID\t"); err != nil {
+			return err
+		}
+	}
+	fmt.Fprintln(tw, "PLUGINS\tLIFESPAN\tACTIVE\tAGE")
+	for _, sess := range ss.Items {
+		if showIDs {
+			_, err := fmt.Fprintf(tw, "%s\t", sess.ID)
+			if err != nil {
+				return err
+			}
+		}
+		_, err := fmt.Fprintf(tw, "%s\t%v\t%v\t%s\n", strings.Join(sess.Plugins, ", "), fmtDuration(time.Duration(sess.Lifespan)), sess.Active(), fmtTime(sess.CreatedAt))
+		if err != nil {
+			return err
+		}
+	}
+	if err := tw.Flush(); err != nil {
+		return err
+	}
+
+	if ss.EndCursor != nil {
+		_, err := fmt.Fprintf(w, "\n\n# Previous page:\n\tcalyptia get trace_sessions --pipeline %s --before %s\n", pipelineID, *ss.EndCursor)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func renderTraceSessionTable(w io.Writer, sess types.TraceSession, showIDs bool) error {
+	tw := tabwriter.NewWriter(w, 0, 4, 1, ' ', 0)
+	if showIDs {
+		if _, err := fmt.Fprint(tw, "ID\t"); err != nil {
+			return err
+		}
+	}
+	fmt.Fprintln(tw, "PLUGINS\tLIFESPAN\tACTIVE\tAGE")
+	if showIDs {
+		_, err := fmt.Fprintf(tw, "%s\t", sess.ID)
+		if err != nil {
+			return err
+		}
+	}
+	_, err := fmt.Fprintf(tw, "%s\t%v\t%v\t%s\n", strings.Join(sess.Plugins, ", "), fmtDuration(time.Duration(sess.Lifespan)), sess.Active(), fmtTime(sess.CreatedAt))
+	if err != nil {
+		return err
+	}
+
+	return tw.Flush()
+}
+
+func (config *config) completeTraceSessions(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	ss, err := config.fetchAllTraceSessions()
+	if err != nil {
+		return nil, cobra.ShellCompDirectiveError
+	}
+
+	if ss == nil {
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	}
+
+	out := make([]string, len(ss))
+	for i, p := range ss {
+		out[i] = p.ID
+	}
+
+	return out, cobra.ShellCompDirectiveNoFileComp
+}
+
+func (config *config) fetchAllTraceSessions() ([]types.TraceSession, error) {
+	pp, err := config.fetchAllPipelines()
+	if err != nil {
+		return nil, err
+	}
+
+	if len(pp) == 0 {
+		return nil, nil
+	}
+
+	var ss []types.TraceSession
+	var mu sync.Mutex
+
+	g, gctx := errgroup.WithContext(config.ctx)
+	for _, pip := range pp {
+		a := pip
+		g.Go(func() error {
+			got, err := config.cloud.TraceSessions(gctx, a.ID, types.TraceSessionsParams{})
+			if err != nil {
+				return err
+			}
+
+			mu.Lock()
```

Comment:
Do we need to lock access to the local variables?

Reply:
Yeah, because the data is fetched concurrently.
We do the same for all of these completions.